### DISCUSSION TITLE
Big project (300 sensors) - help choosing which one + extra questions

### DISCUSSION THREAD

**Discussion by Nej4218** [rbaron/b-parasite#193] [Q&A] [answered]
I’m excited about this project! I’ll begin by outlining my plans and providing some context for the project.

Project Details:
- I will create a plant wall with a sensor in each planter, using a total of 300 sensors. The area is 194 cm (6 feet 4⅜ inches) by 255 cm (8 feet 425⁄64 inches).

- My setup includes an Intel NUC 7 with Zigbee technology, located approximately 9 meters (8 feet 425⁄64 inches) from the plant wall.

- I'm new to the smart home world, including home assistant and programming.

- The planters measure 9.5 cm wide by 8 cm deep (4-inch nursery pot size).

Questions:
- Given the range and need for user-friendly and stable sensors, should I choose Zigbee, BLE, or WiFi sensors?

- Could there be potential issues such as interferences due to the proximity of the sensors? Would this lead to WiFi lag or home assistant issues?

- Are the dimensions of the b-parasite specified?

- Can I adjust the size of the sensors without the need for extensive reprogramming?

- Am I permitted to add my logo in addition to the b-parasite logo? I'm not selling the product, just documenting the smart plant wall project (which i plan on giving a project "logo" for the smart water reservoir/atlas scientific sensors, dosing pumps and all that), ill of course link to this GitHub and so forth - logo attached

![Living wall_20240424_221223_0000](https://github.com/rbaron/b-parasite/assets/168338588/5df1f578-93a8-4341-9083-c8e4156290eb)

- Will the Intel Nuc 7 have issues recieving that amount of data at once?

**Comment by rbaron** [comment, marked as answer]
Hey @Nej4218

That looks like an interesting project. It's not trivial at this scale.

If you allow for some unsolicited advice: I would definitely start way smaller, specially since you mentioned not having a lot of experience. I would also simplify things a bit. For example, having one sensor per row or groups of similar plants. If it's a research project, make sure to read [What to expect from the sensor](https://github.com/rbaron/b-parasite/wiki/What-to-Expect-from-the-Sensor) -- it's going to be challenging to precisely compare values of two different boards anyway unless you at least calibrate each of them individually. Battery management alone at that scale is tricky, and so is handling with inevitable sensor failures and software updates. Tiny battery + wireless is great for spaced out sensors around the house. For 100+ sensors close together in a custom built setup, I would start to consider going wired + shared power.

> Given the range and need for user-friendly and stable sensors, should I choose Zigbee, BLE, or WiFi sensors?
> Could there be potential issues such as interferences due to the proximity of the sensors? Would this lead to WiFi lag or home assistant issues?
> Will the Intel Nuc 7 have issues recieving that amount of data at once?

I have never used anywhere near that many sensors in such close proximity, so I can't say how reliable they would be. I would guess not very, but some of those transmit infrequently enough that there may be very few active at any given time. There are also many parameters to tweak in each of those scenarios, such as transmission interval, power

> Are the dimensions of the b-parasite specified?

The fabrication files are available in the `kicad/` directory.

> Can I adjust the size of the sensors without the need for extensive reprogramming?

See #131 and https://github.com/rbaron/b-parasite/issues/79#issuecomment-1650278703.

> Am I permitted to add my logo in addition to the b-parasite logo? I'm not selling the product, just documenting the smart plant wall project (which i plan on giving a project "logo" for the smart water reservoir/atlas scientific sensors, dosing pumps and all that), ill of course link to this GitHub and so forth - logo attached

Yes, the hardware is released under https://creativecommons.org/licenses/by-sa/4.0/.

**Reply by Nej4218** [reply]
I appreciate the swift reply! 
>If you allow for some unsolicited advice: I would definitely start way smaller, specially since you mentioned not having a lot of experience. I would also simplify things a bit. For example, having one sensor per row or groups of similar plants.

I was planning on acquiring a few premade b-parasites from someone making a bulk order to keep the prices down, asking if they are willing to also preflash while I'm learning (which seems like there is) first to test how it works and play around with integration into HA. If everything works well I'll try to get 22 custom b-parasites for the zones of 12 plants each to actually do it myself 

> If it's a research project, make sure to read What to expect from the sensor -- it's going to be challenging to precisely compare values of two different boards anyway unless you at least calibrate each of them individually.

I am totally fine with the precision of the sensors, it's simply a project I'm doing because I love plants, data and I want to get into programming/home assistant. The sensors purpose will mostly be to adjust approximate watering schedule that fits the majority of the plants as the roots develop, as well as keeping an eye on my slightly more expensive aroids 😅

> - it's going to be challenging to precisely compare values of two different boards anyway unless you at least calibrate each of them individually


I will want try to calibrate all of them, but we'll see haha

> Battery management alone at that scale is tricky, and so is handling with inevitable sensor failures and software updates. Tiny battery + wireless is great for spaced out sensors around the house. For 100+ sensors close together in a custom built setup, I would start to consider going wired + shared power.

My main issue is basically space, i use a plastic planter with drip system into local water reservoirs with wicks, the plants cover an entire wall from side to side with water reservoir just bellow. So batteries would be ideal (imo)

I'd most likely try to adjust it to send data every 30 min or 1 hour to prolong the battery life as much as possible - I'm aiming for them to last around a year if possible, If that works out I dont mind changing the batteries yearly, i can get them in bulk / 200 batteries for ~ 57usd. 
Unless you have to reflash and so forth all over again after changing then it would suck a bit. 

In terms of sensor failure I plan on having a few extra on hand for sure, and software updates are probably unavoidable for any HA integration device i think?*, if i were to go with the alternative Xiaomi flora sensors (which imo suck ∆$$ compared to b-parasite) i would have to deal with similar issues and probably more expensive/less fun

>I have never used anywhere near that many sensors in such close proximity, so I can't say how reliable they would be. I would guess not very, but some of those transmit infrequently enough that there may be very few active at any given time. There are also many parameters to tweak in each of those scenarios, such as transmission interval, power

Great idea about intervals! I'll have to look into that, and I read a bit further and I just have to have adequate amount of Zigbee router/extenders (if i decide to go that route) 

In the end the sensors don't play the most crucial part of the plant wall, but a "nice to have" rather than a "need to have".  The water reservoir on the other hand is where I need pretty spot on readings, but for that I'm going with atlas scientific sensors/probes.

I'll Def keep y'all updated with pics on the community gallery
